### PR TITLE
Update nat_uac_test flag for via re-invite issue

### PR DESCRIPTION
**Pull request by noahmehl**
This is to fixed the issue described here:

https://lists.kamailio.org/pipermail/sr-users/2016-August/094211.html

Deeper explanation here:

https://lists.kamailio.org/pipermail/sr-users/2016-August/094218.html

### CODE REVIEW

**Comment by noahmehl** [comment]
@lazedo, I have updated this to be a variable.  Do you need anything else to accept?

**Comment by sergey-safarov** [comment]
I think this usage configuration is not wide used and may be relocated to `defs.cfg`

**Comment by noahmehl** [comment]
@sergey-safarov I don't understand your comment, the full changes does move this to the defs.cfg?

**Comment by sergey-safarov** [comment]
yes, you settings is already in defs.cfg and redefined from local.cfg
Think in local.cfg this will be unnecessary, because this usage case have small usage. And may redefined/added only on your installation.

**Comment by lazedo** [comment]
@noahmehl this is fine as-is. needs rebase & set the default to 3 in defs.cfg

**Comment by noahmehl** [comment]
@sergey-safarov I understand your comment now.  So I added the trydef in local.cfg, and then if it's not there, the defs.cfg will go ahead and substdef.  Make sense?

@lazedo this should be rebased now, hopefully I didn't break anything.

File: kamailio/defs.cfg [hunk 1]
```diff
@@ -99,6 +99,10 @@
 #!define KZ_PRESENCE_REQUEST_PROBE 1
 #!endif
 
+#!ifndef NAT_UAC_TEST_LEVEL_OVERRIDE
+#!substdef "!NAT_UAC_TEST_LEVEL!3!g"
+#!endif
```
[the remark this answers is not in the document]
Comment (noahmehl) [reply kamailio/defs.cfg]: This should be complete.

File: kamailio/local.cfg [hunk 2]
```diff
@@ -83,6 +83,16 @@
...
 mhomed=0
 
+################################################################################
+## NAT
+################################################################################
+## These parameters are OPTIONAL.
+##    An operator can update the nat_uac_test() flag
+##    level in the nat role.  Need to enable the NAT_UAC_TEST_LEVEL_OVERRIDE to
+##    disable the substdef in defs.conf in addition to the substdef below.
+# # #!trydef NAT_UAC_TEST_LEVEL_OVERRIDE
```
[the remark this answers is not in the document]
Comment (noahmehl) [reply kamailio/local.cfg]: This should be done.

File: kamailio/local.cfg [hunk 3]
```diff
@@ -106,6 +106,12 @@ mhomed=0
 ##!define KZ_UDP_REGISTRAR_PORT 7000
 ##!define KZ_TCP_REGISTRAR_PORT 7000
 
+################################################################################
+## NAT
+################################################################################
+## These parameters are OPTIONAL.
+# # #!substdef "!NAT_UAC_TEST_LEVEL!3!g"
```
Comment (lazedo) [inline kamailio/local.cfg]: sorry for being picky. you can set here the value 1 (it is commented) as an example of overriding the default value. you can also describe that the default value is 3. somethink like
```
this parameter is OPTIONAL
it allows overriding the nat_uac_test with a different value than "3" (default) as it is proven that it fixes the issue reported here (add the url)
```

**Comment by noahmehl** [comment]
Ok, one more try.  I had to fix our repo and add a singe commit with the requested changes.

**Comment by lazedo** [comment]
@noahmehl great & thanks.
you should do the same for the 4.2 PR

**Comment by noahmehl** [comment]
@lazedo will do :)

**Pull request merged**